bson/bsoncodec/default_value_encoders.go

```diff
@@ -349,18 +364,31 @@ func (dve DefaultValueEncoders) ArrayEncodeValue(ec EncodeContext, vw bsonrw.Val
 		return err
 	}
 
-	encoder, err := ec.LookupEncoder(val.Type().Elem())
-	if err != nil {
-		return err
+	encoder, encErr := ec.LookupEncoder(val.Type().Elem())
+	if encErr != nil && val.Type().Elem().Kind() != reflect.Interface {
+		return encErr
 	}
 
 	for idx := 0; idx < val.Len(); idx++ {
+		currVal := val.Index(idx)
+		currEncoder := encoder
+		if currEncoder == nil && (currVal.Kind() == reflect.Interface) {
+			currVal = currVal.Elem()
+			if !currVal.IsValid() {
+				return encErr
+			}
+			currEncoder, err = ec.LookupEncoder(currVal.Type())
```

Comment:
Same comment here about `encErr` as in `mapEncodeValue`.

Reply:
done